### ISSUE TITLE
ICLMetric.compute() silently drops under-gathered documents from both numerator and denominator on IndexError, inflating reported accuracy

### ISSUE THREAD

**Issue by shaurya416** [allenai/OLMo#919]
`olmo/eval/downstream.py:130-149,163` (HEAD `090253dac6688f2532509daa7aa2eb5fae50e956`, current `main`)

```python
for doc_id in loglikelihood_dict:
    # each doc_id might have a different number of continuation
    num_continuations = len(loglikelihood_dict[doc_id].keys())
    loglikelihoods = torch.tensor([-float("inf")] * num_continuations)

    skip_document = False
    for cont_id in loglikelihood_dict[doc_id]:
        try:
            loglikelihoods[cont_id] = loglikelihood_dict[doc_id][cont_id]
        except IndexError:
            # We didn't process all of the continuations, so skip this document.
            skip_document = True
            break

    if skip_document:
        continue
    if self.metric_type in ["ce_loss", "bpb"]:
        correct.append(loglikelihoods[0])  # Only one answer is scored
    else:
        correct.append(1.0 if torch.argmax(loglikelihoods).item() == label_dict[doc_id] else 0.0)
...
score = sum(correct) / len(correct)
```

`num_continuations` is derived from `len(loglikelihood_dict[doc_id].keys())` -- the count of continuation ids actually gathered for that document, not the number the task defines. When gathering is incomplete in a way that leaves a gap in the middle of the id range (plausible under the `DistributedSampler(..., drop_last=False)` padding this same function's docstring already calls out two lines above), the tensor sized from that undercount is too short for a surviving higher `cont_id`, `loglikelihoods[cont_id]` raises `IndexError`, and the whole document is `continue`d out of the loop. It never reaches `correct`, so it is removed from both the numerator and `len(correct)`, the denominator. No `logging.warning`, no counter, nothing in `out` records that N documents were skipped this run.

## What happens (reproduced on the real module)

`ICLMetric.compute()` (`olmo/eval/downstream.py:130-149,163`, verified at HEAD `090253dac6688f2532509daa7aa2eb5fae50e956` = current `main`) sizes each document's `loglikelihoods` tensor from `len(loglikelihood_dict[doc_id].keys())` — the count of continuation ids actually gathered — rather than from the task's declared number of continuations. If a non-trailing `cont_id` is missing for a document, a surviving higher `cont_id` overflows the undersized tensor, raises `IndexError`, and the whole document is `continue`d out of `correct` — removed from both numerator and denominator with no `logging.warning`, no counter, and nothing in the returned metrics dict (`olmo/eval/evaluator.py:compute_metrics`) recording that N documents were skipped. Reproduced against the real, unmodified `ICLMetric` class (imported via `importlib`, not a copy) with the reported numbers matching exactly: 1.0 / 0.6000000238418579 / 1.0 for the control/control/degenerate cases.

Correct the reachability argument: this is not, as the finding states, "plausible under the `DistributedSampler(drop_last=False)` padding" comment two lines above — that comment concerns duplicate over-gathering, which the surrounding dict-construction code already handles. The real, verified trigger is `TrainConfig.eval_subset_num_batches` (`olmo/config.py:1167`, default `-1`/off) combined with the `islice`-based per-rank truncation in `Trainer.eval()` (`olmo/train.py:1024-1038`) acting on a strided (non-shuffled) `DistributedSampler` (`olmo/eval/__init__.py`) — an opt-in but ordinary in-loop-eval configuration, not an exotic one.

Suggested fix stands as written: size `loglikelihoods` from the task's known continuation count rather than the observed key count (so a genuinely missing continuation can't masquerade as an out-of-range index), or keep the skip but count and emit it (`eval/<task>/skipped_docs`) so a shrunk denominator is visible. The cited precedent (#643, #712, #718, all confirmed via `gh pr view` to touch this exact function with counting fixes) supports treating this as a counting bug worth a PR, not a silent-skip design.

Happy to open the PR.
